test_scripts/API/Restructuring_OnResetTimeout/common_OnResetTimeout.lua

```diff
@@ -0,0 +1,1047 @@
+  local consentRPC = "GetInteriorVehicleDataConsent"
+  EXPECT_HMICALL(c.getHMIEventName(consentRPC), c.getHMIRequestParams(consentRPC, pModuleType, pAppId))
+  :Do(function()
+    c.hmiNotification(pRequestID, pMethodName, pResetPeriod)
+    RUN_AFTER(c.hmiNotification, pWait)
+  end)
+  :Do(function(_, _)
+    -- HMI does not respond
+    end)
+  c.getMobileSession(pAppId):ExpectResponse(cid, { success = false, resultCode = "GENERIC_ERROR" })
+end
+
+function c.backupHMICapabilities()
+  local hmiCapabilitiesFile = commonFunctions:read_parameter_from_smart_device_link_ini("HMICapabilities")
+  commonPreconditions:BackupFile(hmiCapabilitiesFile)
+end
+
+local function audibleState(pAppId)
+  if not pAppId then pAppId = 1 end
+  local appParams = config["application" .. pAppId].registerAppInterfaceParams
+  local audibleStateValue
+  if appParams.isMediaApplication == true then
+    audibleStateValue = "AUDIBLE"
+  else
+    audibleStateValue = "NOT_AUDIBLE"
+  end
+  return audibleStateValue
+end
+
+function c.activateApp(pAppId)
+  if not pAppId then pAppId = 1 end
+  local pHMIAppId = c.getHMIAppId(pAppId)
+  local mobSession = c.getMobileSession(pAppId)
+  local requestId = test.hmiConnection:SendRequest("SDL.ActivateApp", { appID = pHMIAppId })
+  EXPECT_HMIRESPONSE(requestId)
+  mobSession:ExpectNotification("OnHMIStatus", { hmiLevel = "FULL", audioStreamingState = audibleState(pAppId),
+      systemContext = "MAIN" })
+  utils.wait()
+end
+
+function c.hmiNotification(pRequestID, pMethodName, pResetPeriod)
+  if not pRequestID then pRequestID = c.getHMIAppId() end
+  if not pResetPeriod then pResetPeriod = 3000 end
+  c.getHMIConnection():SendNotification("BasicCommunication.OnResetTimeout",
+    { requestID = pRequestID,
+    methodName = pMethodName,
+    resetPeriod = pResetPeriod
+  })
+end
+
+function c.sendLocation( pRequestID, pMethodName, pResetPeriod, pWait )
+  local cid = c.getMobileSession():SendRPC("SendLocation", sendLocationRequestParams)
+  EXPECT_HMICALL("Navigation.SendLocation", { appID = c.getHMIAppId() })
+  :Do(function()
+    c.hmiNotification(pRequestID, pMethodName, pResetPeriod)
+    RUN_AFTER(c.hmiNotification, pWait)
+  end)
+  :Do(function(_, data)
+      c.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  c.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+function c.sendLocationError( pRequestID, pMethodName, pResetPeriod, pWait )
+  local cid = c.getMobileSession():SendRPC("SendLocation", sendLocationRequestParams)
+  EXPECT_HMICALL("Navigation.SendLocation", { appID = c.getHMIAppId() })
+  :Do(function()
+    c.hmiNotification(pRequestID, pMethodName, pResetPeriod)
+    RUN_AFTER(c.hmiNotification, pWait)
+  end)
+  :Do(function(_, _)
+    -- HMI does not respond
+  end)
+  c.getMobileSession():ExpectResponse(cid, { success = false, resultCode = "GENERIC_ERROR" })
```

Comment:
Get time of the response receiving and calculate the time from notification to response.